crates/rpc/src/v02/types.rs

```diff
@@ -31,15 +31,42 @@ pub mod request {
         DeployAccount(BroadcastedDeployAccountTransaction),
     }
 
-    // TODO make sure deserialization is not ambiguous between V1 and V2
+    impl BroadcastedTransaction {
+        pub fn into_invoke(self) -> Option<BroadcastedInvokeTransaction> {
+            match self {
+                Self::Invoke(x) => Some(x),
+                _ => None,
+            }
+        }
+
+        pub fn into_declare(self) -> Option<BroadcastedDeclareTransaction> {
+            match self {
+                Self::Declare(x) => Some(x),
+                _ => None,
+            }
+        }
+
+        pub fn into_deploy_account(self) -> Option<BroadcastedDeployAccountTransaction> {
+            match self {
+                Self::DeployAccount(x) => Some(x),
+                _ => None,
+            }
+        }
+
+        #[cfg(test)]
+        pub fn into_invoke_or_panic(self) -> BroadcastedInvokeTransaction {
```

Comment:
nit: I think saving `.upwrap()`s in tests is not worth it.

Reply:
In fact is is exactly same amount of typing: `_or_panic` (9 chars) vs `().unwrap` (9 chars).

Reply:
I got irritated by fmt breaking stuff into so many newlines with those unwraps in the tests but I agree, will remove those `_or_panic` versions.